src/main/java/edu/ucsb/cs156/gauchoride/controllers/RideRequestController.java

```diff
@@ -0,0 +1,132 @@
+package edu.ucsb.cs156.gauchoride.controllers;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import edu.ucsb.cs156.gauchoride.entities.RideRequest;
+import edu.ucsb.cs156.gauchoride.repositories.RideRequestRepository;
+
+import edu.ucsb.cs156.gauchoride.errors.EntityNotFoundException;
+
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+
+import javax.validation.Valid;
+
+@Api(description = "Ride request information (admin, driver, rider)")
+@RequestMapping("/api/riderequests")
+@RestController
+public class RideRequestController extends ApiController {
+    @Autowired
+    RideRequestRepository rideRequestRepository;
+
+    @Autowired
+    ObjectMapper mapper;
+
+    @ApiOperation(value = "Get a list of all ride requests")
+    @PreAuthorize("hasRole('ROLE_ADMIN') || hasRole('ROLE_DRIVER')")
+    @GetMapping("/all")
+    public ResponseEntity<String> rideRequestsAdmin()
+            throws JsonProcessingException {
+        Iterable<RideRequest> rideRequests = rideRequestRepository.findAll();
+        String body = mapper.writeValueAsString(rideRequests);
+        return ResponseEntity.ok().body(body);
+    }
+
+    @ApiOperation(value = "Get a list of all ride requests for current rider")
+    @PreAuthorize("hasRole('ROLE_RIDER')")
+    @GetMapping("/all_rider")
+    public ResponseEntity<String> rideRequestsRider(
+        @ApiParam(name="userid", type="Long", value="userid of the rider", example="123", required=true) @RequestParam Long userid
+        )
+        throws JsonProcessingException {
+
+        Iterable<RideRequest> rideRequests = rideRequestRepository.findAllByUserid(userid);
+        String body = mapper.writeValueAsString(rideRequests);
+        return ResponseEntity.ok().body(body);
+    }
+
+    @ApiOperation(value = "Create a new ride request")
+    @PreAuthorize("hasRole('ROLE_RIDER')")
+    @PostMapping("/post")
+    public RideRequest postRideRequest(
+        @ApiParam(name="userid", type="Long", value="userid of the rider", example="123", required=true)  @RequestParam Long userid,
```

Comment:
Same comment as above.  Do not pass this in the userid, since that any rider to create a rider for another by hacking the backend.  

Instead, get the id of the currently logged in user inside the method, and use that.  That will enforce that a rider can only create a ride with their own userid in the userid field.

There is example code in the proj-courses and proj-happycows code bases that I can refer you to that implements this is; you are welcome to use that code and adapt it to your use case.